### ISSUE TITLE
What is the mechanism of using param 'scale_pos_weight'

### ISSUE THREAD

**Issue by yananchen1989** [dmlc/xgboost#2428]
My dataset has 90% negative samples and 10% positive samples which is very imbalanced.
I try to use the parameter of `scale_pos_weight` and set it as 9. 
What is the mechanism of this param do. 
 I am curious about what it actually means: does it mean repeat the positive samples 9 times? Or everytime pull out 1/9 samples of negative samples and train the model in many times. 
besides, if I have a dataset whose negative samples just a little more than the positive ones, should I need to specify the parameter again?

**Comment by pommedeterresautee** [comment]
Hi,

It doesn't modify your data but change the weights of the positive obeservations.
The approache is called cost sensitive, the idea is to force the model to take care of the rare event by increasing the loss if it fails to correctly predict them.
It works much better than subsampling and co.

https://github.com/dmlc/xgboost/blob/d581a3d0e7fbc8e73dbeb6f81fdc1e101606737e/src/objective/regression_obj.cc#L69-L76

Kind regards,
Michaël

**Comment by MartinThoma** [comment]
I came here looking for the [documentation of `scale_pos_weight`](http://xgboost.readthedocs.io/en/latest/parameter.html):

> Control the balance of positive and negative weights, useful for unbalanced classes. A typical value to consider: sum(negative cases) / sum(positive cases) See Parameters Tuning for more discussion. Also see Higgs Kaggle competition demo for examples: R, py1, py2, py3

I found it via [What is the proper usage of scale_pos_weight in xgboost for imbalanced datasets?](https://stats.stackexchange.com/q/243207/25741).
